src/classes/Path2D.js

```diff
@@ -39,6 +39,7 @@ export default class Path2D {
       throw new TypeError(
         "Failed to execute 'addPath' on 'Path2D': parameter 1 is not of type 'Path2D'."
       );
-    this._path = this._path.concat(path._path);
+    for (let i = 0; i < path._path.length; i++)
+      this._path.push(path._path[i]);
```

Comment:
It doesn't seem to make any difference?

Reply:
its faster : )

Reply:
I think we do not consider the perf of language style, compile will do it.^_^

Reply:
you mean babel should optimize this? bad news:

```js
// jest-canvas-mock/lib/classes/Path2D.js
    this._path = this._path.concat(path._path);
```

its a micro optimization, but then why not

Reply:
`concat` is more simple for code.

Reply:
```js
Array.prototype._pushArray = function _pushArray(a) {
  for (let i = 0; i < a.length; i++)
    this.push(a[i]);
  //return this;
};

this._path._pushArray(path._path);
```

`this._path.push(...path._path);` would be limited by call stack size

Reply:
I still think this is what the compiler should do, and the upper code should not deal with it.

Reply:
![image](https://user-images.githubusercontent.com/7856674/107210726-82826680-6a3f-11eb-839e-b4116ff80c2b.png)

many `concat` usage.

Reply:
> I still think this is what the compiler should do

compiler? the babel compiler or the JS runtime?
some solutions are generally slower (concat + reassign, functional loops, immutable data) 
and are hard to optimize in the runtime

the pragmatic reason for "no" is: its just a micro optimization, with almost zero gain

Reply:
Sorry, I mean v8, not babel.

Reply:
![image](https://user-images.githubusercontent.com/7856674/108960921-9a99ed00-76b1-11eb-8960-57080a1d1bb3.png)